src/client/components/SocialButtons.tsx

```diff
@@ -47,75 +59,94 @@ const iconOverrides = css`
 );
 
-export const SocialButtons = ({
+const SocialButton = ({
+  label,
+  icon,
+  socialProvider,
   queryParams,
-  marginTop,
+  showGap,
+  isNativeApp,
 }: SocialButtonProps) => {
   return (
-    <div css={containerStyles(marginTop)}>
+    <>
       <LinkButton
         priority="tertiary"
         cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgGoogleBrand />}
+        icon={icon}
         href={buildUrlWithQueryParams(
           '/signin/:social',
           {
-            social: 'google',
+            social: socialProvider,
           },
           queryParams,
         )}
-        data-cy="google-sign-in-button"
-        data-link-name="google-social-button"
+        data-cy={`${socialProvider}-sign-in-button`}
+        data-link-name={`${socialProvider}-social-button`}
       >
-        Google
-      </LinkButton>
-      <Gap />
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={buttonOverrides}
-        icon={<SvgFacebookBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'facebook',
-          },
-          queryParams,
-        )}
-        data-cy="facebook-sign-in-button"
-        data-link-name="facebook-social-button"
-      >
-        Facebook
-      </LinkButton>
-      <Gap />
-      <LinkButton
-        priority="tertiary"
-        cssOverrides={[buttonOverrides, iconOverrides]}
-        icon={<SvgAppleBrand />}
-        href={buildUrlWithQueryParams(
-          '/signin/:social',
-          {
-            social: 'apple',
-          },
-          queryParams,
-        )}
-        data-cy="apple-sign-in-button"
-        data-link-name="apple-social-button"
-      >
-        Apple
+        {socialButtonLabel(label, isNativeApp)}
       </LinkButton>
+      {showGap && <Gap isNativeApp={isNativeApp} />}
+    </>
+  );
+};
+
+const socialButtonLabel = (label: string, isNativeApp?: IsNativeApp) => {
+  const capitalisedLabel = label.charAt(0).toUpperCase() + label.slice(1);
+  if (isNativeApp) {
+    return `Continue with ${capitalisedLabel}`;
+  } else {
+    return capitalisedLabel;
+  }
+};
+
+export const SocialButtons = ({
+  queryParams,
+  marginTop,
+  isNativeApp,
+}: SocialButtonsProps) => {
+  let buttonOrder;
+  if (isNativeApp === 'ios') {
+    buttonOrder = ['apple', 'facebook', 'google'];
+  } else if (isNativeApp === 'android') {
+    buttonOrder = ['google', 'facebook', 'apple'];
+  } else {
+    buttonOrder = ['google', 'facebook', 'apple'];
+  }
+  return (
+    <div css={containerStyles(marginTop, isNativeApp)}>
+      {buttonOrder.map((socialProvider, index) => (
+        <SocialButton
+          key={socialProvider}
+          label={socialProvider}
+          icon={
+            socialProvider === 'google' ? (
+              <SvgGoogleBrand />
+            ) : socialProvider === 'apple' ? (
+              <SvgAppleBrand />
+            ) : (
+              <SvgFacebookBrand />
+            )
```

Comment:
this is a bit hard to read, i suggest something similar to the `getButtonOrder` there should be a method which takes a social provider and returns an icon

the default shouldn't be facebook either, the default should probably be no icon

Reply:
`<></>` is fine!